coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/CoralToTrinoSqlCallTransformersUtil.java

```diff
@@ -0,0 +1,247 @@
+    sqlCallTransformerList.add(createSignatureBasedConditionSqlCallTransformer(
+        linkedInFunctionToCoralOperator("com.linkedin.jobs.udf.hive.ConvertIndustryCode"), 1, "converttoindustryv1"));
+    addLinkedInFunctionSignature(linkedInFunctionIdSet, "com.linkedin.jobs.udf.hive.ConvertIndustryCode", 1);
+    sqlCallTransformerList.add(createSignatureBasedConditionSqlCallTransformer(
+        linkedInFunctionToCoralOperator("com.linkedin.stdudfs.urnextractor.hive.UrnExtractorFunctionWrapper"), 1,
+        "urn_extractor"));
+    addLinkedInFunctionSignature(linkedInFunctionIdSet,
+        "com.linkedin.stdudfs.urnextractor.hive.UrnExtractorFunctionWrapper", 1);
+    sqlCallTransformerList.add(createSignatureBasedConditionSqlCallTransformer(
+        linkedInFunctionToCoralOperator("com.linkedin.stdudfs.hive.daliudfs.UrnExtractorFunctionWrapper"), 1,
+        "urn_extractor"));
+    addLinkedInFunctionSignature(linkedInFunctionIdSet,
+        "com.linkedin.stdudfs.hive.daliudfs.UrnExtractorFunctionWrapper", 1);
+
+    initializeLinkedInFunctionTransformerFromHiveRegistry(sqlCallTransformerList, linkedInFunctionIdSet);
+  }
+
+  private static SqlOperator hiveToCoralOperator(String functionName) {
+    Collection<Function> lookup = HIVE_REGISTRY.lookup(functionName);
+    // TODO: provide overloaded function resolution
+    return lookup.iterator().next().getSqlOperator();
+  }
+
+  private static void addLinkedInFunctionSignature(Set<String> linkedInFunctionSignatureSet,
+      String linkedInFunctionName, int operandNum) {
+    linkedInFunctionSignatureSet.add(linkedInFunctionName + "_" + operandNum);
+  }
+
+  private static void initializeLinkedInFunctionTransformerFromHiveRegistry(
+      List<SqlCallTransformer> sqlCallTransformerList, Set<String> linkedInFunctionSignatureSet) {
+    ImmutableMultimap<String, Function> registry = HIVE_REGISTRY.getRegistry();
+    Converter<String, String> caseConverter = CaseFormat.UPPER_CAMEL.converterTo(CaseFormat.LOWER_UNDERSCORE);
+    for (Map.Entry<String, Function> entry : registry.entries()) {
+      // we cannot use entry.getKey() as function name directly, because keys are all lowercase, which will
+      // fail to be converted to lowercase with underscore correctly
+      final String hiveFunctionName = entry.getValue().getFunctionName();
+      if (!hiveFunctionName.startsWith("com.linkedin")) {
+        continue;
+      }
+      String[] nameSplit = hiveFunctionName.split("\\.");
+      // filter above guarantees we've at least 2 entries
+      String className = nameSplit[nameSplit.length - 1];
+      String funcName = caseConverter.convert(className);
+      SqlOperator op = entry.getValue().getSqlOperator();
+      for (int i = op.getOperandCountRange().getMin(); i <= op.getOperandCountRange().getMax(); i++) {
+        if (!linkedInFunctionSignatureSet.contains(funcName + "_" + i)) {
```

Comment:
This is not aligned with the above `linkedInFunctionSignatureSet.add(linkedInFunctionName + "_" + operandNum);`
`linkedInFunctionName` is camel case, while `funcName` is lowercase + underscore.
So it will overwrite the previous registered linkedin UDFs like `com.linkedin.dali.udf.watbotcrawlerlookup.hive.WATBotCrawlerLookup` and cause regression.

Reply:
Good catch. Done.